File: cifrado/escritorio/codigo/Cifrados Kotlin/src/Texto_plano/Texto_plano.kt
package Texto_plano

class Texto_plano:Texto.Texto {

    private var texto:String = "";

    constructor(texto:String){
        this.texto = texto;
    }

    override fun numero_caracteres(): Int {
        return this.texto.length;
    }

    override fun obtener_texto(): String {
        return this.texto;
    }

    override fun agregar_grupos_caracteres(grupo_caracteres: String) {
        this.texto+=grupo_caracteres;
    }

    override fun caracter_en_posicion(posicion: Int): Char {
        return this.texto.get(posicion);
    }

    override fun posicion_de_caracter(caracter: Char): Int {
        return this.texto.indexOf(caracter);
    }

    override fun hay_caracter_en_posicion(posicion: Int): Boolean {
        return posicion<this.numero_caracteres();
    }

    override fun agregar_caracter(caracter: Char) {
        this.texto+=caracter;
    }

    override fun eliminar_ultimo_caracter() {
        this.texto = this.texto.substring(0,numero_caracteres()-1);
    }

    override fun texto_en_grupos(grupos: Int): ArrayList<String> {
        var texto_en_grupos_respuesta:ArrayList<String> = ArrayList();
        var grupo_auxiliar = "";

        var iterador = this.texto.iterator();
        while (iterador.hasNext()){
            grupo_auxiliar+=iterador.next();
            if(grupo_auxiliar.length==grupos){
                texto_en_grupos_respuesta.add(grupo_auxiliar);
                grupo_auxiliar = "";
            }
        }

        return texto_en_grupos_respuesta;
    }
}

File: cifrado/web/codigo/Cifrados C#/clases/Texto_plano/Texto_plano.cs
using System;
using System.Collections;

namespace Cifrado_cesar_poo_desktop.Texto_plano
{
    public class Texto_plano : Texto.Texto
    {
        private String texto;

        public Texto_plano(String texto)
        {
            this.texto = texto;
        }

        public void agregar_texto(String texto){
            this.texto = texto;
        }

        public void agregar_caracter(char caracter)
        {
            this.texto += caracter;
        }

        public void agregar_grupos_caracteres(string grupo_caracteres)
        {
            this.texto += grupo_caracteres;
        }

        public char caracter_en_posicion(int posicion)
        {
            return this.texto[posicion];
        }

        public void eliminar_ultimo_caracter()
        {
            if (this.texto.Length > 0)
            {
                this.texto = this.texto.Substring(0, this.texto.Length - 1);
            }

        }

        public bool hay_caracter_en_posicion(int posicion)
        {
            return posicion < this.texto.Length;
        }

        public int numero_caracteres()
        {
            return this.texto.Length;
        }

        public string obtener_texto()
        {
            return this.texto;
        }

        public int posicion_de_caracter(char caracter)
        {
            return this.texto.IndexOf(caracter);
        }

        public ArrayList texto_en_grupos(int grupos)
        {
            String grupos_caracteres = "";
            ArrayList respuesta_en_grupos = new ArrayList();

            foreach (char caracter in this.texto)
            {
                grupos_caracteres += caracter;

                if (grupos_caracteres.Length == grupos)
                {
                    respuesta_en_grupos.Add(grupos_caracteres);
                    grupos_caracteres = "";
                }
            }

            return respuesta_en_grupos;
        }
    }
}
